funcx_web_service/routes/funcx.py

```diff
@@ -841,23 +843,27 @@ def register_endpoint_2(user: User, user_uuid: str):
     except Exception as e:
         app.logger.debug("Caught random error : {}".format(e))
         response = {'status': 'error',
-                    'reason': f'Caught error while registering endpoint {e}'}
-
-    try:
-        forwarder_ip = app.config['FORWARDER_IP']
-        response = register_with_hub(
-                f"http://{forwarder_ip}:8080", endpoint_uuid, endpoint_ip_addr)
-        app.logger.debug(f"Successfully registered {endpoint_uuid} with forwarder")
+                    'reason': f'Caught error while registering endpoint - {e}'}
 
-    except Exception as e:
-        app.logger.debug("Caught error during forwarder initialization")
-        app.logger.error(e, exc_info=True)
-        response = {'status': 'error',
-                    'reason': f'Failed during broker start {e}'}
+    # A response will have been set by now if there was already an error,
+    # so we should only proceed to try registering with the hub if there
+    # have been no errors so far
+    if response is None:
+        try:
+            forwarder_ip = app.config['FORWARDER_IP']
+            response = register_with_hub(
+                    f"http://{forwarder_ip}:8080", endpoint_uuid, endpoint_ip_addr)
+            app.logger.debug(f"Successfully registered {endpoint_uuid} with forwarder")
 
-    if 'meta' in request.json and endpoint_uuid:
-        ingest_endpoint(user.username, user_uuid, endpoint_uuid, request.json['meta'])
-        app.logger.debug(f"Ingested endpoint {endpoint_uuid}")
+        except Exception as e:
+            app.logger.debug("Caught error during forwarder initialization")
```

Comment:
Consider replacing calls to debug and error here with a call to `logger.exception` which is equivalent to calling `logger.error` with the exception info added in. Keep in mind that `logger.exception` should be called only from the exception handling block.

Reply:
Replaced some `logger.debug` calls with `logger.exception`